Applications/spire/source/charting/chart_view.cpp

```diff
@@ -211,7 +212,6 @@ void ChartView::update_origins() {
   auto y_value = m_bottom_right.m_y - (m_bottom_right.m_y % m_y_axis_step) +
     m_y_axis_step;
   m_y_axis_values.clear();
-  m_x_origin = INT_MAX;
```

Comment:
Why was this removed?

Reply:
If the chart is zoomed in close enough it's possible that there won't be any y-axis labels to display, in that case m_x_origin keeps the last value it had.

Reply:
Fixing that one corner case breaks every other case. Find a way to handle that situation without breaking existing behavior.